features/features_test.go

```diff
@@ -378,59 +378,91 @@ func (s *featureSuite) TestAll(c *C) {
 	defer st.Unlock()
 	tr := config.NewTransaction(st)
 
+	// Use the first feature as a testing feature and overwrite all its values.
+	fakeFeature := features.SnapdFeature(features.NumberOfFeatures())
+	is, why := fakeFeature.IsSupported()
+	c.Check(is, Equals, true)
+	c.Check(why, Equals, "")
+
+	restore := features.MockKnownFeaturesImpl(func() []features.SnapdFeature {
+		features := make([]features.SnapdFeature, 1)
+		features[0] = fakeFeature
+		return features
+	})
+	defer restore()
+
+	features.FeatureNames[fakeFeature] = "fake-feature"
+	defer func() {
+		delete(features.FeatureNames, fakeFeature)
+	}()
+
 	allFeaturesInfo := features.All(tr)
 
+	c.Assert(len(allFeaturesInfo), Equals, 1)
+
 	// Feature flags are included even if value unset
-	layoutsInfo, exists := allFeaturesInfo[features.Layouts.String()]
+	fakeFeatureInfo, exists := allFeaturesInfo[fakeFeature.String()]
 	c.Assert(exists, Equals, true)
 	// Feature flags are supported even if no callback defined.
-	c.Check(layoutsInfo.Supported, Equals, true)
+	c.Check(fakeFeatureInfo.Supported, Equals, true)
 	// Feature flags have a value even if unset.
-	c.Check(layoutsInfo.Enabled, Equals, true)
+	c.Check(fakeFeatureInfo.Enabled, Equals, false)
+
+	var fakeSupported bool
+	var fakeReason string
+	fakeCallback := func() (bool, string) {
+		return fakeSupported, fakeReason
+	}
+	features.FeaturesSupportedCallbacks[fakeFeature] = fakeCallback
+	defer func() {
+		delete(features.FeaturesSupportedCallbacks, fakeFeature)
+	}()
 
 	// Feature flags with defined supported callbacks work correctly.
 
 	// Callbacks which return false result in Supported: false
-	restore := systemd.MockSystemdVersion(229, nil)
-	defer restore()
+	fakeSupported = false
+	fakeReason = "foo"
 	allFeaturesInfo = features.All(tr)
-	quotaGroupsInfo, exists := allFeaturesInfo[features.QuotaGroups.String()]
+	fakeFeatureInfo, exists = allFeaturesInfo[fakeFeature.String()]
```

Comment:
Could be nice to check that `allFeaturesInfo` only has one element, but this is maybe overkill

Reply:
Ideally we'd have more than one feature in the mocked `KnownFeatures` and thus we could ensure that all the fake features are included and no others. But again I think this is overkill